### PR TITLE
Update the plugin

### PR DESCRIPTION
**Pull request by GoWang**
Hi @ispyropoulos,

I am working on the updates with @JasonMrX. 

Basically what we did are:

- Update the plugin with the assets from the newest released version v8.4.8
- Use the minified js instead of the regular one
- Rearrange the assets a bit (more consistent naming, easier for users to install and configurate)
- New scss to help override the flag image paths in intlTelInput_path_override.scss
- Update README accordingly.

We are new to the area and hope that what we did is following the convention. Any suggestion would be appreciated. 

Thanks,

Rong

### CODE REVIEW

**Comment by ispyropoulos** [comment]
I would suggest keeping the same file naming, as per the original library (e.g. `app/assets/javascripts/intlTelInput.js`) and also avoid using the minified versions. There's no need. This is supposed to be a wrapper gem intended for Rails, which minifies javascripts assets by default on production. So, let's keep the normal versions in this repo as it is much easier to track changes between commits.

Also, regarding the CSS part, I would suggest a change. Let's use the `.css` [file](https://github.com/jackocnr/intl-tel-input/tree/master/build/css), instead of the `.scss` file (used in master), as we don't really know if all people use Sass in their stack. We can then remove [this](https://github.com/ispyropoulos/intl-tel-input-rails/blob/19e7335417f199f67783ddbd37b5a23ce345b415/intl-tel-input-rails.gemspec#L22) dependency. Also, please keep the original file naming (e.g. `app/assets/stylesheets/intlTelInput.scss`).

[Here](http://guides.rubyonrails.org/asset_pipeline.html#customizing-the-pipeline) you can find more details about Rails asset compression.

**Comment by GoWang** [comment]
Hi,

Great suggestions. While I totally agree with your first point, I am still confused about the second. First of all you suggest using the `@import` syntax in README file, which I guess is already the SASS taste. Secondly, IMHO people tend to use `@import` rather than `*=require` for a reason [here](https://github.com/rails/sass-rails#important-note).

I was hoping to get some clarification for the second point, but anyway I will make the changes as you suggested. Thanks

Rong

**Comment by ispyropoulos** [comment]
The `@import "intlTelInput"` stated in the README file will have to change as well - I neglected to mention this, sorry. It is currently correct, as the gem depends on `saas-rails`. I am suggesting to drop this dependency as per my previous comment. If we drop it, then we need to use the `require` syntax. I hope that this makes sense. Thanks.

**Comment by GoWang** [comment]
Hi,
I have made the changes. I also updated the README file. Let me know if there is anything else.
Thanks,
Rong

File: README.md [hunk 1]
```diff
@@ -23,15 +23,30 @@ Or install it yourself as:
 
 Add the following directive to your Javascript manifest file:
 
-    //= require intlTelInput
+    //= require intl-tel-input-rails
```
Comment (ispyropoulos) [inline README.md]: You don't need to change this or line 30. The file name is `intlTelInput` (same as before).

**Comment by ispyropoulos** [comment]
A few comments.

1. The stylesheet file in your branch has a Saas extension `.scss` for some reason. We need it to be `.css`. As per my previous comment make sure you are using [this](https://github.com/jackocnr/intl-tel-input/tree/master/build/css) file - obviously you need the contents of that particular file that corresponds to this version of the plugin you are updating. I am guessing you have used the files contained in version [v8.4.7](https://github.com/jackocnr/intl-tel-input/releases/tag/v8.4.7) package, right?
1. I am not sure why you have added file `intlTelInput_path_override`. There is no need to override anything. The only thing you have to do is make a slight change in the `intlTelInput.css` file. After you get the fresh version from the plugin repo, you need to replace the two flag image locations `url("../img/flags.png")` and `url("../img/flags@2x")` with `image-url("flags.png")` and `image-url("flags@2x.png")` respectively, which use the Rails asset pipeline helpers for the stylesheets. If you pay closer attention to this existing version, you will notice that they are already there.
1. Finally, I would like to ask you. Do you changes actually work? Have you tried your branch in a sample Rails project? Make sure that plugin loads correctly with no problems and that it works exactly as the original plugin. I will give it a go before merging anyway, but please make sure that there are no problems.

**Comment by GoWang** [comment]
Hi,

1. I will make sure the plugin is up to date with v8.4.9 in the next commit. I used `.scss` because you stated in your previous comment that "Also, please keep the original file naming (e.g. `app/assets/stylesheets/intlTelInput.scss`).". I just got a bit confused here. I though as long as the contents are not `sass-rails` dependent, the file extension is negligible. But that's fine I will change it back to `css` as you wish. 

2. Firstly, I thought [`image-url`](https://github.com/rails/sass-rails#asset-helpers) helper is `sass-rails` dependent, but you wanna drop the dependency actually. I was thinking that changing it to `background-image:url(/assets/flags.png)` makes more sense right? 
The other thing is, the initial intention for overriding the css in another file is to facilitate the gem updating, so that each time the plugin is updated, we could just copy and paste the new assets files without further changes. But it might confuse the gem users though, so I will remove the overriding file and apply the changes in the original file.

3. I've tested in two rails apps. One with `sass-rails` and the other one without, and the both work as expected. I will test it again after the new changes. Thanks for the reminding.

Rong

**Comment by ispyropoulos** [comment]
Hi Rong,

1. When I wrote _"keep the original file naming"_ I meant the file naming of the original javascript plugin.

2. The `image-url` helper works for both css and saas. Regarding you other point, you are right that is allows easier updates. However, it is just a couple of easy replacements, which are done once - by the person that contributes in this gem wrapper. This way we also use a Rails capability, which allows for smaller and cleaner code.

3. Nice!

**Comment by GoWang** [comment]
Hi Ilias,

The  changes are almost done in @JasonMrX's last commit `c76e258`, except for the `image-url` helper.  We found some documents in [here](https://github.com/rails/sass-rails#asset-helpers) and [here](http://guides.rubyonrails.org/asset_pipeline.html#css-and-sass), which seem to (we are not sure) indicate that the `image-url` helper are in sass flavor. Moreover, we tried but we could not make the `image-url` work without changing the `css` extension to `scss`, which is indeed in sass flavor. Therefore we keep it as `url(/assets/flag.png)`, which works as expected. 

As I stated, we've tested the updated gem in actual rails apps with or without `sass-rails`. Please advise the above problem if you have any idea. Thanks.

Rong

**Comment by ispyropoulos** [comment]
Yes, it seems that this is the case. It also seems that Sass comes as the default choice in Rails project. Therefore, there's no really a benefit from dropping the sass-rails dependency [here](https://github.com/ispyropoulos/intl-tel-input-rails/blob/master/intl-tel-input-rails.gemspec#L22). So let's do a few final things.

1. Please reinstate the `sass-rails` dependency.
2. Put back the `.scss` extension.
3. Use the `image-url` helper.
4. Put the `@import "intlTelInput";` syntax in the README.md.

After doing the above, please squash your commits and we are ready to merge.

**Comment by JasonMrX** [comment]
Hi,

We've squashed the changes. Please have a look. Thanks.

Jason

**Pull request merged**